packages/web/src/components/search-bar/SearchBarResult.tsx

```diff
@@ -1,118 +1,136 @@
+  to: string
+}) => (
+  <Link to={to}>
+    <Flex
+      alignItems='center'
+      gap='s'
+      p='s'
+      css={{
+        minWidth: 0,
+        '&:hover': {
+          backgroundColor: 'var(--neutral-light-8)'
+        }
+      }}
+    >
+      {children}
+    </Flex>
+  </Link>
+)
+
+type ResultTextProps = {
+  primary: string
+  secondary: string
+  badges?: React.ReactNode
+}
 
-import searchBarStyles from './SearchBar.module.css'
-import styles from './SearchBarResult.module.css'
+const ResultText = ({ primary, secondary, badges }: ResultTextProps) => (
+  <Flex direction='column' css={{ minWidth: 0, flex: 1 }}>
+    <Flex alignItems='center' gap='2xs' css={{ minWidth: 0 }}>
+      <Text
+        variant='body'
+        size='s'
+        color='default'
+        css={{ minWidth: 0 }}
+        ellipses
+      >
+        {primary}
+      </Text>
+      {badges}
+    </Flex>
+    <Text
+      variant='body'
+      size='xs'
+      color='subdued'
+      css={{ minWidth: 0 }}
+      ellipses
+    >
+      {secondary}
+    </Text>
+  </Flex>
+)
 
-type ImageProps = {
-  defaultImage: string
-  artwork: Record<SquareSizes, string>
-  size: SquareSizes
-  isUser: boolean
+type UserResultProps = {
+  user: User
 }
 
-const Image = memo(({ defaultImage, artwork, size, isUser }: ImageProps) => {
-  const image = useImageSize({
-    artwork,
-    targetSize: size,
-    defaultImage,
-    preloadImageFn: preload
-  })
-  return (
-    <DynamicImage
-      skeletonClassName={cn({ [styles.userImageContainerSkeleton]: isUser })}
-      wrapperClassName={cn(styles.imageContainer)}
-      className={cn({
-        [styles.image]: image,
-        [styles.userImage]: isUser,
-        [styles.emptyUserImage]: isUser && image === defaultImage
-      })}
-      image={image}
+const UserResult = ({ user }: UserResultProps) => (
+  <ResultWrapper to={profilePage(user.handle)}>
+    <Avatar
+      h={30}
+      w={30}
+      src={user.profile_picture?.['150x150'] || imageProfilePicEmpty}
+      borderWidth='thin'
+      css={{ flexShrink: 0 }}
+    />
+    <ResultText
+      primary={user.name}
+      secondary={`@${user.handle}`}
+      badges={<UserBadges userId={user.user_id} badgeSize={12} inline />}
     />
-  )
-})
+  </ResultWrapper>
+)
 
-type SearchBarResultProps = {
-  kind: Kind
-  userId: number
-  primary: string
-  secondary?: string
-  artwork: Record<SquareSizes, string> | null
-  size: SquareSizes | null
-  defaultImage: string
-  isVerifiedUser?: boolean
-  tier: BadgeTier
+type TrackResultProps = {
+  track: UserTrackMetadata
 }
 
-const SearchBarResult = memo(
-  ({
-    kind,
-    userId,
-    primary,
-    secondary,
-    artwork,
-    size,
-    defaultImage,
-    isVerifiedUser = false,
-    tier
-  }: SearchBarResultProps) => {
-    const isUser = kind === Kind.USERS
+const TrackResult = ({ track }: TrackResultProps) => (
+  <ResultWrapper
+    to={`/${track.user.handle}/tracks/${track.title}-${track.track_id}`}
```

Comment:
feels like we should have a util somewhere to build this. Worth using/adding to routeUtils?

Reply:
yup def, handling